packages/edit-widgets/src/components/widget-area/index.js

```diff
@@ -9,13 +16,35 @@ import {
 } from '@wordpress/block-editor';
 import { withDispatch, withSelect } from '@wordpress/data';
 
+const { mediaUpload } = mediaUtils;
+
 function WidgetArea( {
 	blockEditorSettings,
 	blocks,
 	initialOpen,
 	updateBlocks,
 	widgetAreaName,
+	hasUploadPermissions,
 } ) {
+	const settings = useMemo(
```

Comment:
Is `useMemo` necessary here? Spreading the `settings` object to contain a new `__experimentalMediaUpload` property doesn't seem very expensive.

Reply:
Yes, it is not very expensive computing the settings object itself. The reason I am using it is that without it each time WidgetArea renders settings would point to a different reference that would trigger a rerender BlockEditorProvider with new settings which then would need to update the block editor store with new settings and probably cause other rerenders.